src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,61 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int STARTING_CAPACITY = 10;
+    private K[] keys = (K[]) new Object[STARTING_CAPACITY];
+    private V[] values = (V[]) new Object[STARTING_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex == -1) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            values[keyIndex] = value;
+        }
+        if (keys.length == size) {
+            increaseCapacity();
+        }
```

Comment:
```suggestion
        if (keys.length == size) {
            increaseCapacity();
        }
```
we shouldn`t have this condition